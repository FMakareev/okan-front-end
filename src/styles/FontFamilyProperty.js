/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
