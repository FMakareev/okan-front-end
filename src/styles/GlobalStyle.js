import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
