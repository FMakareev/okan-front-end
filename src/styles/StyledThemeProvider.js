import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
