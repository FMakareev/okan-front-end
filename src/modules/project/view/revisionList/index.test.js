import React from 'react';
import renderer from 'react-test-renderer';

import { BrowserRouter } from 'react-router-dom';
import { RevisionListPage } from './index';
import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';

test('RevisionListPage: рендер без ошибок', () => {
  const output = renderer.create(
    <StyledThemeProvider>
      <BrowserRouter>
        <RevisionListPage />
      </BrowserRouter>
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});
