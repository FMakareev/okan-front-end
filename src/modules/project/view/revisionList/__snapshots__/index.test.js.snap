// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevisionListPage: рендер без ошибок 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 wterhv-0 clExyh"
  >
    <div
      className="sc-1jxwge6-0 sc-14k9gvj-0 bfeFrq"
      width="100%"
    >
      <table
        className="cqxezr-0 xj0wna-0 gjgxNz"
        width="100%"
      >
        <tr
          className="sc-1cxzlk0-0 imvHLU"
        >
          <th
            className="sc46x0-0 kpDrKL"
            color="color11"
            fontFamily="primary500"
            fontSize={6}
            width="100px"
          >
            № Ревизии
          </th>
          <th
            className="sc46x0-0 cklyzK"
            color="color11"
            fontFamily="primary500"
            fontSize={6}
            width="500px"
          >
            Ф. И. О.
          </th>
          <th
            className="sc46x0-0 fGAmgj"
            color="color11"
            fontFamily="primary500"
            fontSize={6}
          >
            Дата
          </th>
          <th
            className="sc46x0-0 fGAmgj"
            color="color11"
            fontFamily="primary500"
            fontSize={6}
          />
        </tr>
        <tbody
          className="obix2s-0 XmuzM"
        >
          <tr
            className="sc-1cxzlk0-0 imvHLU"
          >
            <td
              className="ivgi64-0 xj0wna-1 kUjLEN"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              1
            </td>
            <td
              className="ivgi64-0 xj0wna-1 cruUmi"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              Циалковский Святослав Валентинович
            </td>
            <td
              className="ivgi64-0 xj0wna-1 cruUmi"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              12.11.2018
            </td>
            <td
              className="ivgi64-0 xj0wna-1 hwlvCK"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              <div
                className="sc-1jxwge6-0 wterhv-0 dPBtpW"
              >
                <button
                  className="mxzgdr-0 fLHGuY"
                >
                  <svg
                    fill="none"
                    height="30"
                    viewBox="0 0 30 30"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="white"
                      height="29.5"
                      rx="4.75"
                      stroke="#848484"
                      stroke-width="0.5"
                      width="29.5"
                      x="0.25"
                      y="0.25"
                    />
                    <path
                      d="M9.6015 14.5299C10.4955 14.1372 11.3507 13.9407 12.1665 13.9407H23V11.9555C23 11.1946 22.7135 10.5408 22.1403 9.99492C21.5673 9.44903 20.8818 9.17613 20.0831 9.17613H12.9997V8.77912C12.9997 8.01816 12.7133 7.36472 12.1404 6.81896C11.5675 6.27307 10.8817 6 10.0831 6H5.91661C5.11801 6 4.43224 6.27303 3.8593 6.81896C3.28636 7.36477 3 8.01821 3 8.77916V20.6896C3 20.7228 3.00223 20.7746 3.00657 20.845C3.01085 20.9152 3.01313 20.967 3.01313 21L3.07817 20.926L7.46615 16.0125C7.99563 15.4169 8.70744 14.9226 9.6015 14.5299Z"
                      fill="#848484"
                    />
                    <path
                      d="M25.692 16.1629C25.4867 16.0544 25.2522 16 24.9885 16H12.3372C11.8256 16 11.2694 16.148 10.6687 16.4439C10.0679 16.7401 9.60082 17.1 9.26758 17.5253L5.36056 22.4749C5.12019 22.7665 5 23.0416 5 23.3001C5 23.55 5.10276 23.7292 5.30815 23.8373C5.51354 23.9458 5.74801 24 6.01164 24H18.6633C19.1749 24 19.7308 23.852 20.3317 23.5563C20.9324 23.2605 21.3993 22.9001 21.7327 22.4749L25.6397 17.5253C25.88 17.2336 26 16.9588 26 16.7001C26.0002 16.4503 25.8976 16.2712 25.692 16.1629Z"
                      fill="#848484"
                    />
                  </svg>
                </button>
                <button
                  className="mxzgdr-0 fLHGuY"
                >
                  <svg
                    fill="none"
                    height="30"
                    viewBox="0 0 30 30"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="white"
                      height="29.5"
                      rx="4.75"
                      stroke="#848484"
                      stroke-width="0.5"
                      width="29.5"
                      x="0.25"
                      y="0.25"
                    />
                    <path
                      d="M7.40039 11.9998H11.1004C11.3766 11.9998 11.6004 12.2236 11.6004 12.4998V19.1362C11.6004 19.4124 11.8243 19.6362 12.1004 19.6362H16.7005C16.9767 19.6362 17.2005 19.4124 17.2005 19.1362V12.4998C17.2005 12.2236 17.4244 11.9998 17.7005 11.9998H21.4006L14.4005 5.99969L7.40039 11.9998Z"
                      fill="#848484"
                      stroke="#848484"
                      stroke-width="0.25"
                    />
                    <path
                      d="M22.3002 21H6.5C6.22386 21 6 21.2239 6 21.5V23.5C6 23.7762 6.22386 24 6.5 24H22.3002C22.5764 24 22.8002 23.7762 22.8002 23.5V21.5C22.8002 21.2239 22.5764 21 22.3002 21Z"
                      fill="#848484"
                      stroke="#848484"
                      stroke-width="0.25"
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className="sc-1cxzlk0-0 imvHLU"
          >
            <td
              className="ivgi64-0 xj0wna-1 kUjLEN"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              1
            </td>
            <td
              className="ivgi64-0 xj0wna-1 cruUmi"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              Циалковский Святослав Валентинович
            </td>
            <td
              className="ivgi64-0 xj0wna-1 cruUmi"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              12.11.2018
            </td>
            <td
              className="ivgi64-0 xj0wna-1 eLdBSp"
              color="#333333"
              fontFamily="primary300"
              fontSize="18px"
            >
              <div
                className="sc-1jxwge6-0 wterhv-0 dPBtpW"
              >
                <button
                  className="mxzgdr-0 fLHGuY"
                >
                  <svg
                    fill="none"
                    height="30"
                    viewBox="0 0 30 30"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="white"
                      height="29.5"
                      rx="4.75"
                      stroke="#848484"
                      stroke-width="0.5"
                      width="29.5"
                      x="0.25"
                      y="0.25"
                    />
                    <path
                      d="M9.6015 14.5299C10.4955 14.1372 11.3507 13.9407 12.1665 13.9407H23V11.9555C23 11.1946 22.7135 10.5408 22.1403 9.99492C21.5673 9.44903 20.8818 9.17613 20.0831 9.17613H12.9997V8.77912C12.9997 8.01816 12.7133 7.36472 12.1404 6.81896C11.5675 6.27307 10.8817 6 10.0831 6H5.91661C5.11801 6 4.43224 6.27303 3.8593 6.81896C3.28636 7.36477 3 8.01821 3 8.77916V20.6896C3 20.7228 3.00223 20.7746 3.00657 20.845C3.01085 20.9152 3.01313 20.967 3.01313 21L3.07817 20.926L7.46615 16.0125C7.99563 15.4169 8.70744 14.9226 9.6015 14.5299Z"
                      fill="#848484"
                    />
                    <path
                      d="M25.692 16.1629C25.4867 16.0544 25.2522 16 24.9885 16H12.3372C11.8256 16 11.2694 16.148 10.6687 16.4439C10.0679 16.7401 9.60082 17.1 9.26758 17.5253L5.36056 22.4749C5.12019 22.7665 5 23.0416 5 23.3001C5 23.55 5.10276 23.7292 5.30815 23.8373C5.51354 23.9458 5.74801 24 6.01164 24H18.6633C19.1749 24 19.7308 23.852 20.3317 23.5563C20.9324 23.2605 21.3993 22.9001 21.7327 22.4749L25.6397 17.5253C25.88 17.2336 26 16.9588 26 16.7001C26.0002 16.4503 25.8976 16.2712 25.692 16.1629Z"
                      fill="#848484"
                    />
                  </svg>
                </button>
                <button
                  className="mxzgdr-0 fLHGuY"
                >
                  <svg
                    fill="none"
                    height="30"
                    viewBox="0 0 30 30"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="white"
                      height="29.5"
                      rx="4.75"
                      stroke="#848484"
                      stroke-width="0.5"
                      width="29.5"
                      x="0.25"
                      y="0.25"
                    />
                    <path
                      d="M7.40039 11.9998H11.1004C11.3766 11.9998 11.6004 12.2236 11.6004 12.4998V19.1362C11.6004 19.4124 11.8243 19.6362 12.1004 19.6362H16.7005C16.9767 19.6362 17.2005 19.4124 17.2005 19.1362V12.4998C17.2005 12.2236 17.4244 11.9998 17.7005 11.9998H21.4006L14.4005 5.99969L7.40039 11.9998Z"
                      fill="#848484"
                      stroke="#848484"
                      stroke-width="0.25"
                    />
                    <path
                      d="M22.3002 21H6.5C6.22386 21 6 21.2239 6 21.5V23.5C6 23.7762 6.22386 24 6.5 24H22.3002C22.5764 24 22.8002 23.7762 22.8002 23.5V21.5C22.8002 21.2239 22.5764 21 22.3002 21Z"
                      fill="#848484"
                      stroke="#848484"
                      stroke-width="0.25"
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
